### PR TITLE
Symlink worktree for discovered shell tests

### PR DESCRIPTION
**Pull request by psss**
Instead of copying again the whole directory tree for discovered
shell tests, let's create a symlink to the plan worktree which
already contains everything needed for testing. This saves disk
space and allows to share content between prepare and execute, for
example prepare step can compile binaries which are then used by
tests. Test coverage extended to cover this scenario.

### CODE REVIEW

**Review by thrix — APPROVED** [review]
LGTM

File: tests/run/worktree/main.fmf [hunk 1]
```diff
@@ -2,9 +2,10 @@ summary: Check whether worktree is correctly set
 description:
     For the `prepare` and `finish` steps, the current working
     directory must be set to the metadata tree root for best user
-    experience. By default tested in a `container, for pull
-    requests also against `local` and during `full` testing using
-    `virtual` as well.
+    experience. Also tests defined directly in the plan (under the
+    `discover` or `execute` step) should be executed there. By
+    default tested in a `container, for pull requests also against
```
[the remark this answers is not in the document]
Comment (psss) [reply tests/run/worktree/main.fmf]: Thanks for catching this. Fixed.

**Comment by lukaszachy** [comment]
Testsuite fails because of https://github.com/psss/tmt/issues/954

**Comment by psss** [comment]
Sorry for the force-push noise here. We're checking with @lukaszachy why the new distgit test was skipped...

**Comment by psss** [comment]
All test failures are the weird empty environment file issue, the rest is green, merging.

**Pull request merged**